### PR TITLE
CJ4, CRJ, reorganized buttons, MobiFlight dependency

### PR DESCRIPTION
**Pull request by elewarr**
**Just wanted to share with my changes. This is work in progress. Use if you want, close if you want or keep open to see future changes.**

- Cessna CJ4 configuration added
- Bombardier CRJ configuration (WIP)
- reorganized buttons/encoders order (see READMEs)
- MobiFlight dependent data (for CJ4)
- blinking gear when transition in progress
- blinking yaw damper when gear is down
- blinking flaps when gear is up
- blinking battery when discharging (not reliable, Cj4 gives positive load on discharge, other planes - negative)
- moved jinja2 stuff to separate files
- use index definitions instead of hardcoded values
- READMEs added

### CODE REVIEW

File: Configurations/Functions/Generic/ap-spd-vs-encoder-switcher.jinja2 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {% if data.get_simvar_value('AUTOPILOT_VERTICAL_HOLD') == 1.0 %}
-    {{ data.trigger_encoder_alternate(5, True) }}
-    {{ data.set_encoder_led(5, True) }}
+    {{ data.trigger_encoder_alternate(4, True) }}
```
Comment (maartentamboer) [inline Configurations/Functions/Generic/ap-spd-vs-encoder-switcher.jinja2:2]: With the new feature of initializing variables from the config file we could move the encoder index to a variable. This makes sure this generic function is truly generic. Currently this change breaks the default config file
```jinja2
{% set encoder_index = get_global_variable('AUTOPILOT_ENCODER') %}
{% if data.get_simvar_value('AUTOPILOT_VERTICAL_HOLD') == 1.0 %}
    {{ data.trigger_encoder_alternate(encoder_index, True) }}
```

```json
"initialization": {
    "global_variables": [
      {
        "name": "AUTOPILOT_ENCODER",
        "value": 4
      }
    ]
  }
```

Comment (maartentamboer) [reply Configurations/Functions/Generic/ap-spd-vs-encoder-switcher.jinja2:2]: Same of course applies for other functions that have been added to the generic folder

File: Configurations/config_default.json [hunk 2]
```diff
@@ -9,12 +9,6 @@
   "encoders": [
```
Comment (maartentamboer) [inline Configurations/config_default.json:9]: I see that there are quite a few changes in the default config file. Can you make a small list of what's changed, then when this get's merged and released I can include it in the release notes.

File: Configurations/Functions/Generic/ap-spd-vs-encoder-switcher.jinja2 (hunk 1 above)
Comment (maartentamboer) [reply Configurations/Functions/Generic/ap-spd-vs-encoder-switcher.jinja2:2]: > Currently this change breaks the default config file

Nevermind, just saw that the default config file was changed as well

**Review by maartentamboer — COMMENTED** [review]
Wow really nice. I would love to merge this. Let me know what you think of the proposed changes to the Generic functions.

**Comment by elewarr** [comment]
I'll try to describe buttons/encoders setup later, when I'm back home. For now just the picture.
image

**Comment by elewarr** [comment]
Added definitions for buttons/encoders indices but these are untested as of yet. I'll test it when I'm back home.

**Review by maartentamboer — APPROVED** [review]

**Comment by elewarr** [comment]
List of changes in 1st comment.

**Review by maartentamboer — APPROVED** [review]

**Comment by maartentamboer** [comment]
Merged! Thanks for contributing 👍

**Comment by maartentamboer** [comment]
Released: https://github.com/maartentamboer/X-Touch-Mini-FS2020/releases/tag/v1.6.0

**Pull request merged**